core/math/a_star_grid_2d.cpp

```diff
@@ -598,7 +609,8 @@ void AStarGrid2D::_bind_methods() {
 	ADD_PROPERTY(PropertyInfo(Variant::VECTOR2, "cell_size"), "set_cell_size", "get_cell_size");
 
 	ADD_PROPERTY(PropertyInfo(Variant::BOOL, "jumping_enabled"), "set_jumping_enabled", "is_jumping_enabled");
-	ADD_PROPERTY(PropertyInfo(Variant::INT, "default_heuristic", PROPERTY_HINT_ENUM, "Euclidean,Manhattan,Octile,Chebyshev,Max"), "set_default_heuristic", "get_default_heuristic");
+	ADD_PROPERTY(PropertyInfo(Variant::INT, "default_compute_heuristic", PROPERTY_HINT_ENUM, "Euclidean,Manhattan,Octile,Chebyshev,Max"), "set_default_compute_heuristic", "get_default_compute_heuristic");
+	ADD_PROPERTY(PropertyInfo(Variant::INT, "default_estimate_heuristic", PROPERTY_HINT_ENUM, "Euclidean,Manhattan,Octile,Chebyshev,Max"), "set_default_estimate_heuristic", "get_default_estimate_heuristic");
 	ADD_PROPERTY(PropertyInfo(Variant::INT, "diagonal_mode", PROPERTY_HINT_ENUM, "Never,Always,At Least One Walkable,Only If No Obstacles,Max"), "set_diagonal_mode", "get_diagonal_mode");
```

Comment:
These hints shouldn't include `Max`, that's invalid value (it's just for the checks).
```suggestion
	ADD_PROPERTY(PropertyInfo(Variant::INT, "default_compute_heuristic", PROPERTY_HINT_ENUM, "Euclidean,Manhattan,Octile,Chebyshev"), "set_default_compute_heuristic", "get_default_compute_heuristic");
	ADD_PROPERTY(PropertyInfo(Variant::INT, "default_estimate_heuristic", PROPERTY_HINT_ENUM, "Euclidean,Manhattan,Octile,Chebyshev"), "set_default_estimate_heuristic", "get_default_estimate_heuristic");
	ADD_PROPERTY(PropertyInfo(Variant::INT, "diagonal_mode", PROPERTY_HINT_ENUM, "Never,Always,At Least One Walkable,Only If No Obstacles"), "set_diagonal_mode", "get_diagonal_mode");
```

It's not staightforward to inspect `AStarGrid2D` in the inspector and make these hints be used at all but it's possible:
```gdscript
@tool
extends EditorScript

var astar := AStarGrid2D.new()

func _run():
	get_editor_interface().inspect_object(astar, "", true)
```
![Godot_v4 0-beta10_win64_0r2dBrKhht](https://user-images.githubusercontent.com/9283098/209407615-1c9e8eba-588b-41c2-b507-c2238aa62b78.png)
![Godot_v4 0-beta10_win64_m5ZCiTWPrW](https://user-images.githubusercontent.com/9283098/209407616-fbed19a0-46de-4b94-b807-25375914c6c8.png)

Reply:
Ah, indeed it's incorrect. Thanks for pointing me!

Reply:
Oh, seems like I forgot to mentioned that yesterday: I also don't think the `(int)` casts in `ERR_FAIL_INDEX` are needed. At least I don't think I've seen them around the codebase. They are included in the printed error as seen above so yeah, probably not wanted. That's not really an issue but maybe you could change these in your other PR.

Reply:
@kleonc Then I suppose you should check and approve this - https://github.com/godotengine/godot/pull/70485 and I will include that fix into it

Reply:
NVM, I think I will create a separate PR to remove all occurences of `ERR_FAIL_INDEX((int)` in the engine